ci/ReleaseTool/Common.cs

```diff
@@ -55,5 +57,108 @@ private static string AppendDirectorySeparator(string originalPath)
 
             return originalPath + directorySeparator;
         }
+
+        public static bool IsMarkdownHeading(string markdownLine, int level, string startTitle = null)
+        {
+            var heading = $"{new string('#', level)} {startTitle ?? string.Empty}";
+
+            return markdownLine.StartsWith(heading);
+        }
+
+        public static bool UpdateChangeLog(string ChangeLogFilePath, string Version, GitClient gitClient, string ChangeLogReleaseHeadingTemplate)
+        {
+            using (new WorkingDirectoryScope(gitClient.RepositoryPath))
+            {
+                if (File.Exists(ChangeLogFilePath))
+                {
+                    var originalContents = File.ReadAllText(ChangeLogFilePath);
+                    var changelog = File.ReadAllLines(ChangeLogFilePath).ToList();
+                    var releaseHeading = string.Format(ChangeLogReleaseHeadingTemplate, Version,
+                        DateTime.Now);
+                    var releaseIndex = changelog.FindIndex(line => IsMarkdownHeading(line, 2, $"[`{Version}`] - "));
+                    // If we already have a changelog entry for this release, replace it.
+                    if (releaseIndex != -1)
```

Comment:
Why does `releaseIndex != -1` mean the heading exists? I thought it would return the index of the line which matches our predicate

Reply:
Returns -1 when the heading e.g. "[0.10.0] -" does not exist, right?

Reply:
I am blind